static/app/views/profiling/landingAggregateFlamegraph.tsx

```diff
@@ -52,6 +60,37 @@ function decodeViewOrDefault(
   return defaultValue;
 }
 
+interface SlowestFlamegraphFrame {
+  aggregate: {
+    selfWeight: number;
+    totalWeight: number;
+  };
+  frame: FlamegraphFrame['frame'];
+  instances: CallTreeNode[];
+}
+
+function sortByTotalWeight(a: SlowestFlamegraphFrame, b: SlowestFlamegraphFrame) {
+  return b.aggregate.totalWeight - a.aggregate.totalWeight;
+}
+
+function getSlowestFlamegraphFrames(flamegraph: Flamegraph): SlowestFlamegraphFrame[] {
+  const results: Map<FlamegraphFrame['frame'], SlowestFlamegraphFrame> = new Map();
+
+  for (const frame of flamegraph.frames) {
```

Comment:
I forget but does this flamegraph update to only contain application/system frames the user selects it?

Reply:
Yes it does. Tbh if we want that tbh, I could see it being useful, but then we also usually also have dedicated controls for this...